animation_nodes/nodes/matrix/distribute_matrices.pyx

```diff
@@ -260,9 +265,19 @@ class DistributeMatricesNode(bpy.types.Node, AnimationNode):
         stepSin = sin(angleStep)
 
         for i in range(amount):
-            vector.x = iCos * radius
-            vector.y = iSin * radius
-            vector.z = 0
+            if self.planeAxis == "XY":
+                vector.x = iCos * radius
+                vector.y = iSin * radius
+                vector.z = 0
+            elif self.planeAxis == "YZ":
+                vector.x = 0
+                vector.y = iSin * radius
+                vector.z = iCos * radius
+            else:
+                vector.x = iCos * radius
+                vector.y = 0
+                vector.z = iSin * radius
+
             setTranslationMatrix(matrices.data + i, &vector)
             setMatrixCustomZRotation(matrices.data + i, iCos, iSin)
```

Comment:
Should this change as well?

Reply:
Omar, If I define `setMatrixCustom...` according to the [right-hand rule](https://en.wikipedia.org/wiki/Rotation_matrix) then they do not output correct rotation,
```
cdef inline void setMatrixCustomXRotation(Matrix4* m, double iCos, double iSin):
    m.a22 = m.a33 = iCos
    m.a23, m.a32 = -iSin, iSin

cdef inline void setMatrixCustomYRotation(Matrix4* m, double iCos, double iSin):
    m.a11 = m.a33 = iCos
    m.a13, m.a31 = iSin, -iSin
```
![Screenshot from 2021-02-12 00-05-19](https://user-images.githubusercontent.com/30294746/107682353-3d239a80-6cc6-11eb-9b0c-986ab3e0e4aa.png)



However, if I change the -ve sign then they output correct rotation,
```
cdef inline void setMatrixCustomXRotation(Matrix4* m, double iCos, double iSin):
    m.a22 = m.a33 = iCos
    m.a23, m.a32 = -iSin, iSin

cdef inline void setMatrixCustomYRotation(Matrix4* m, double iCos, double iSin):
    m.a11 = m.a33 = iCos
    m.a13, m.a31 = -iSin, iSin
```
![Screenshot from 2021-02-12 00-06-32](https://user-images.githubusercontent.com/30294746/107682393-4b71b680-6cc6-11eb-8105-4ce1b3dbb0bf.png)

Is there something that I'm doing wrong?

Reply:
@3DSinghVFX Those functions don't really have anything to do with rotations matrices or the right hand rule. You construct the rotation of the matrix by making three decisions:

- What should the local x axis of the matrix be aligned with? In the XY case, we choose to align with the normal to the circle. The normal to the circle is obviously `V(cos, sin, 0)` by looking at the unit circle.
- What should the local y axis of the matrix be aligned with? In the XY case, we choose to align with the tangent to the circle. The tangent to the circle is perpendicular to the normal, which is equivalent to a `pi/2` degree rotation, which we can compute through trigonometric identifies as follows, `V(cos + pi/2, sin + pi/2, 0)` -> `V(-sin, cos, 0)`.
- What should the local z axis of the matrix be aligned with? In the XY case, we choose to align with `V(0, 0, 1)`.

To actually construct the matrix from all of those vectors, you assign them to the 3x3 columns of the matrix. The x-aligned vector is set to the first column, the y-aligned vector is set to the second column, and the z-aligned vector is set to the their column. In the XY case, it looks like this:

```
cos  -sin 0
sin  cos  0
0    0    1
```

The other options can be deduced similarly.

Reply:
Thanks for the explanation. I have aligned the axis in the cycle: `Y` in `XY`, `Z` in `YZ`, and `X` in `ZX`.